app/components/custom_status/clear_button.tsx

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+import CompassIcon from '@components/compass_icon';
+import {Theme} from '@mm-redux/types/preferences';
+
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {preventDoubleTap} from '@utils/tap';
+import {TouchableOpacity} from 'react-native';
+
+interface Props {
+    handlePress: () => void;
+    size?: number;
+    theme: Theme;
+    testID?: string;
+    iconName: string,
+}
+
+const ClearButton = (props: Props) => {
+    // Note: We need a default iconName as the defaultProps
+    // don't get applied initially when the value is undefined
```

Comment:
@enahum I think that the defaults did not work, because the `iconName` prop ist not defined as optional. I never tried it, but I am pretty sure in this case a `defaultProp` will not get handled 🧐

Reply:
It could be.. have not tried it either BUT i do doubt it as React will asign values at runtime not really pay attention to TS, right?

So what is really going on here? And why the need to assign a default value during props deconstruction when TS is saying that it must be defined.